### PR TITLE
ci: Update k8s to use only runtimeclass

### LINKED ISSUE #1527: update k8s to use only runtimeclass

### PR DESCRIPTION
**Pull request by GabyCT**
With the update of k8s to version 1.14.1, we can deprecate the
use of untrusted annotations.

Depends-on: github.com/kata-containers/runtime#1618

Fixes #1527

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/kubernetes/init.sh [hunk 1]
```diff
@@ -81,14 +78,12 @@ sleep_time=5
...
-	echo "Install RuntimeClass resource definition"
-	kubectl apply -f \
-		"https://raw.githubusercontent.com/kubernetes/kubernetes/v${kubernetes_version/-*}/cluster/addons/runtimeclass/runtimeclass_crd.yaml"
-	echo "Create kata RuntimeClass resource"
-	kubectl create -f "${runtimeclass_files_path}/kata-runtimeclass.yaml"
-fi
+runtimeclass_files_path="${SCRIPT_PATH}/runtimeclass_workloads"
+echo "Install RuntimeClass resource definition"
+kubectl apply -f \
+	"https://raw.githubusercontent.com/kubernetes/kubernetes/v${kubernetes_version/-*}/cluster/addons/runtimeclass/runtimeclass_crd.yaml"
```
Comment (krsna1729) [inline integration/kubernetes/init.sh]: This CRD is now a standard api. This URL does not exist anymore.

File: .ci/install_kubernetes.sh [hunk 2]
```diff
@@ -39,9 +34,7 @@ chronic sudo -E apt install kubernetes-cni=0.6.0-00
...
 chronic sudo -E apt install --allow-downgrades -y kubelet="$kubernetes_version" kubeadm="$kubernetes_version" kubectl="$kubernetes_version"
 
-if [ "${use_runtime_class}"  == true ]; then
-	kubelet_systemd_file="/etc/systemd/system/kubelet.service.d/10-kubeadm.conf"
-	feature_gate="--feature-gates RuntimeClass=true"
-	echo "Configure Kubelet service to enable RuntimeClass feature"
-	sudo sed -i "s/ExecStart=\/.*$/& $feature_gate/" "$kubelet_systemd_file"
-fi
+kubelet_systemd_file="/etc/systemd/system/kubelet.service.d/10-kubeadm.conf"
+feature_gate="--feature-gates RuntimeClass=true"
```
Comment (krsna1729) [inline .ci/install_kubernetes.sh]: With 1.14 RuntimeClass is beta. So it is enabled by default. We can remove the feature-gate everywhere

File: integration/kubernetes/init.sh (hunk 1 above)
Comment (GabyCT) [reply integration/kubernetes/init.sh]: @krsna1729 changes applied

File: .ci/install_kubernetes.sh (hunk 2 above)
Comment (GabyCT) [reply .ci/install_kubernetes.sh]: krsna1729 changes applied

File: integration/kubernetes/init.sh [hunk 3]
```diff
@@ -52,11 +51,9 @@ kubeadm_config_file="$(mktemp --tmpdir kubeadm_config.XXXXXX.yaml)"
...
 sed -i "s|KUBERNETES_VERSION|v${kubernetes_version/-*}|" "${kubeadm_config_file}"
 
-if [ "${use_runtime_class}"  == true ]; then
-	echo "Add RuntimeClass feature for apiserver in kubeadm config file"
-	echo "apiServerExtraArgs:" >> "${kubeadm_config_file}"
-	echo "  feature-gates: RuntimeClass=true" >> "${kubeadm_config_file}"
-fi
+echo "Add RuntimeClass feature for apiserver in kubeadm config file"
+echo "apiServerExtraArgs:" >> "${kubeadm_config_file}"
+echo "  feature-gates: RuntimeClass=true" >> "${kubeadm_config_file}"
```
Comment (krsna1729) [inline integration/kubernetes/init.sh]: can remove these steps too

File: integration/kubernetes/init.sh [hunk 4]
```diff
@@ -81,14 +78,9 @@ sleep_time=5
...
-	runtimeclass_files_path="${SCRIPT_PATH}/runtimeclass_workloads"
-	echo "Install RuntimeClass resource definition"
-	kubectl apply -f \
-		"https://raw.githubusercontent.com/kubernetes/kubernetes/v${kubernetes_version/-*}/cluster/addons/runtimeclass/runtimeclass_crd.yaml"
-	echo "Create kata RuntimeClass resource"
-	kubectl create -f "${runtimeclass_files_path}/kata-runtimeclass.yaml"
-fi
+runtimeclass_files_path="${SCRIPT_PATH}/runtimeclass_workloads"
+echo "Create kata RuntimeClass resource"
+kubectl create -f "${runtimeclass_files_path}/kata-runtimeclass.yaml"
```
Comment (krsna1729) [inline integration/kubernetes/init.sh]: This yaml needs to be updated like here https://github.com/clearlinux/cloud-native-setup/blob/master/clr-k8s-examples/8-kata/kata-qemu-runtimeClass.yaml#L2-L6

File: integration/kubernetes/init.sh (hunk 3 above)
Comment (GabyCT) [reply integration/kubernetes/init.sh]: @krsna1729 changes applied

File: integration/kubernetes/init.sh (hunk 4 above)
Comment (GabyCT) [reply integration/kubernetes/init.sh]: done

File: integration/kubernetes/runtimeclass_workloads/kata-runtimeclass.yaml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 kind: RuntimeClass
-apiVersion: node.k8s.io/v1alpha1
+apiVersion: node.k8s.io/v1beta1
 metadata:
   name: kata
 spec:
```
Comment (krsna1729) [inline integration/kubernetes/runtimeclass_workloads/kata-runtimeclass.yaml]: one thing missed - 
```
spec:
  runtimeHandler: kata
```
becomes
```
handler: kata
```

Comment (GabyCT) [reply integration/kubernetes/runtimeclass_workloads/kata-runtimeclass.yaml]: @krsna1729 done

File: .ci/configure_crio_for_kata.sh [hunk 6]
```diff
@@ -11,22 +11,9 @@ set -o pipefail
...
-
-if [ "${use_runtime_class}"  == true ]; then
-	echo "Configure runtimes map for RuntimeClass feature"
-	echo "- Set runc as default runtime"
-	sudo sed -i 's!runtime_path =.*!runtime_path = "/usr/local/bin/crio-runc"!' "$crio_config_file"
-	echo "- Add kata-runtime to the runtimes map"
-	sudo sed -i '/crio-runc/a[crio.runtime.runtimes.kata]' "$crio_config_file"
-	sudo sed -i '/kata/aruntime_path = "/usr/local/bin/kata-runtime"' "$crio_config_file"
-else
-	echo "Configure runtimes for trusted/untrusted annotations"
```
Comment (chavafg) [inline .ci/configure_crio_for_kata.sh]: I think that we will still need to check for cri-o version and configure it depending the version as we still use cri-o 1.10 for openshift v3.10, which does not support runtimeclass.

Comment (GabyCT) [reply .ci/configure_crio_for_kata.sh]: @chavafg changes applied

Comment (GabyCT) [reply .ci/configure_crio_for_kata.sh]: @chavafg if I update the cri-o version we have a lot of failures in the cri-o tests :S

Comment (GabyCT) [reply .ci/configure_crio_for_kata.sh]: @chavafg doing the installation of CRI-O locally with 1.14.0, I get the following error
```
Get CRI Tools
can't load package: package github.com/kubernetes-sigs/cri-tools: no Go files in /home/popular/go/src/github.com/kubernetes-sigs/cri-tools
/home/popular/go/src/github.com/kubernetes-sigs/cri-tools /home/popular/go/src/github.com/cri-o/cri-o /home/popular/go/src/github.com/kata-containers/tests/.ci
error: pathspec 'v1.14.0' did not match any file(s) known to git.
```

Comment (GabyCT) [reply .ci/configure_crio_for_kata.sh]: but if I put the version v1.13.6 the installation is done succesfully

Comment (grahamwhaley) [reply .ci/configure_crio_for_kata.sh]: Ah, thanks for the clarification on the openshift stuff @chavafg - I think that is an element I was missing in my understanding :-)

**Comment by GabyCT** [comment]
This PR needs https://github.com/kata-containers/runtime/pull/1618 (the CI for that PR is already successful)

**Comment by GabyCT** [comment]
Adding the DNM label until we merged first the PR of runtime

**Comment by GabyCT** [comment]
@grahamwhaley looks like WIP-check is not working

**Comment by grahamwhaley** [comment]
@GabyCT - seems to be running again https://zuul.opendev.org/t/kata-containers/builds?job_name=kata-github-wip
Maybe there was a queue or temporary downtime.

File: integration/kubernetes/nginx.bats [hunk 7]
```diff
@@ -32,7 +32,7 @@ setup() {
 	busybox_pod="test-nginx"
 	kubectl run $busybox_pod --restart=Never --image="$busybox_image" \
 		-- wget --timeout=5 "$deployment"
-	cmd="kubectl get pods -a | grep $busybox_pod | grep Completed"
+	cmd="kubectl get pods | grep $busybox_pod | grep Completed"
```
Comment (grahamwhaley) [inline integration/kubernetes/nginx.bats]: hey @GabyCT - did you also sneak a bugfix in here - as I just found `kubectl get pods -a` is not a valid `kubectl` cmd :-)
Probably should have dropped that on a separate commit in this PR.
Was this an effect of k8s updating, or has it always been broken? If broken, how did it ever get through the CI/bats ?
thx!

Comment (GabyCT) [reply integration/kubernetes/nginx.bats]: @grahamwhaley it was an effect of k8s updating to a newer version

**Pull request merged**